apps/examples/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,459 @@
+			}
+
+			for (int wr_i = 0; wr_i < (file_size / 1024); wr_i++ ) {
+				ret = write(fd, rw_buf, 512);
+				if (ret != (512)) {
+					printf("Unable to create  file: %s,size = %d ,ret = %d\n", file[0], file_size, ret);
+					close(fd);
+					return ERROR;
+				}
+			}
+
+			for (int buf_size = 512; buf_size <= (b_sz); buf_size *= 2) {
+				tt = 0;
+				if (file_size < buf_size)
+					continue;
+
+				frt_fd = open("/dev/timer0", O_RDONLY);
+				if (frt_fd < 0) {
+					printf("Unable to do open file : /dev/timer0, errno : %d\n", file[0], errno);
+					return ERROR;
+				}
+
+				ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN);
+				ioctl(frt_fd, TCIOC_START, TRUE);
+				ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before);
+
+				for (int o_loop = 1; o_loop <= MAX_ITR; o_loop++) {
+					fd = open(file[0], O_RDOK);
+					if (fd < 0) {
+						printf("Unable to open file %s to read  fd : %d \n", file[0], fd);
+						close(frt_fd);
+						return ERROR;
+					}
+
+					for (int i_loop = 1; i_loop <= file_size / buf_size; i_loop++) {
+						ret = read(fd, rw_buf, buf_size);
+						if (ret != (buf_size)) {
+							printf("Unable to read file: %s, buf_size = %d , f_size = %d ret = %d\n", file[0], buf_size, file_size, ret);
+							close(frt_fd);
+							close(fd);
+							return ERROR;
+						}
+					}
+					close(fd);
+				}
+
+				ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+				ioctl(frt_fd, TCIOC_STOP, 0);
+				close(frt_fd);
+
+				tt += (after.timeleft - before.timeleft);
+
+				printf("End File size :%d bytes Buf size : %d bytes tt : %lu us\n",file_size, buf_size, tt);
+			}
+		}
+		break;
+
+	case TEST_FLAG_GC_EFFECT :
+		memset(rw_buf, 0, BUFFER_SIZE);
+		tt = tt2 = 0;
+
+		for (i = 0; i < 10; i++) {
+			frt_fd = open("/dev/timer0", O_RDONLY);
+			if (frt_fd < 0) {
+				printf("Unable to do open file : /dev/timer0, errno : %d\n", file[0], errno);
+				return ERROR;
+			}
+			ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN);
+			ioctl(frt_fd, TCIOC_START, TRUE);
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before);
+
+			fd = open(file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file : %s , fd: %d\n", file[0], fd);
+				close(frt_fd);
+				return ERROR;
+			}
+
+			ret = write(fd, rw_buf, b_sz);
+			if (ret != (b_sz)) {
+				 printf("Unable to write to file: %s,ret = %d\n", file[0], ret);
+				 close(frt_fd);
+				 close(fd);
+				 return ERROR;
+			}
+			close(fd);
+
+			ret = unlink(file[0]);
+			if (ret != OK) {
+				printf("Failed to unlink file %s, ret : %d, errno : %d\n", file[0], ret, errno);
+				close(frt_fd);
+				return ERROR;
+			}
+
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			ioctl(frt_fd, TCIOC_STOP, 0);
+			close(frt_fd);
+
+			tt2 = after.timeleft - before.timeleft;
+			printf("iteration time: %lu us\n", i, tt2);
+			tt += tt2;
+		}
+		printf("\nFinal time = %lu us\n", tt);
+		break;
+
+	default:
+		break;
+	}
+
+	return ret;
+}
+
```

Comment:
In this function: -
1. If you want to keep the initial declaration of I and j, then use them maximum wherever possible. If you need a 3rd iterator only then declare new.
2. If you don't keep those declarations then you can use whatever names you want as per the TC wr_itr/b_sz/f_sz, etc.
3. Is it possible to club some part of code in this function? Eachcase has different test code but declarations, timer setup and starting, buffer generation might be common, can we club that?